### ISSUE TITLE
Change how `AsyncMagicMethodViolation` works

### ISSUE THREAD

**Issue by sobolevn** [wemake-services/wemake-python-styleguide#718]
After creating #717 I got and idea that `AsyncMagicMethodViolation` (implemented by awesome @bellum in #684) also works incorrectly. Let me show an example of what's wrong:

```python
def CustomClass(CustomInterface):
     async def __protocol__(self):
          ...
```

In this case some other library defines that custom magic method `__protocol__` should be `async`. And we will raise a violation for that. It is a false-positive.

We should work with magic methods blacklist instead of whitelist. 
Things to refactor:
1. Create a blacklist of all magic methods that cannot be used as `async`: like `__new__`, `__init__`, `__str__`, etc
2. Rewrite docs about this violation, state that custom magic methods can be sync or async 
3. Change tests
